.github/workflows/auto_deploy_staging.yaml

```
@@ -0,0 +1,32 @@
+name: Apply changes
+
+# only trigger on pull request closed events
+on:
+  pull_request:
+    types: [ closed ]
+    branches:
+      - staging
+
+jobs:
+  terraform_apply:
+    if: github.event.pull_request.merged == true
```

```suggestion
    if: github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged == true
```

I played around a bit, and for me this is what works. I have no idea why a more specific check works when a less specific one doesn't...